quickwit/quickwit-search/src/leaf.rs

```diff
@@ -424,31 +530,78 @@ pub async fn leaf_search(
     searcher_context: Arc<SearcherContext>,
     request: &SearchRequest,
     index_storage: Arc<dyn Storage>,
-    splits: &[SplitIdAndFooterOffsets],
+    splits: &mut [SplitIdAndFooterOffsets],
     doc_mapper: Arc<dyn DocMapper>,
 ) -> Result<LeafSearchResponse, SearchError> {
     let request = Arc::new(request.clone());
-    let leaf_search_single_split_futures: Vec<_> = splits
+
+    let split_filter = if request.sort_by_field.is_none() {
```

Comment:
```suggestion
    // The leaf search code contains some logic that makes it possible to skip entire splits
    // when we are confident they won't make it into top K.
    // To make this optimization as potent as possible, we sort the splits so that the first splits
    // are the most likely to fill our Top K.
    let split_filter = if request.sort_by_field.is_none() {
```